dali/pipeline/util/bounding_box_utils.h

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.c++ copyswa
+
+#ifndef DALI_PIPELINE_UTIL_BOUNDING_BOX_UTILS_H_
+#define DALI_PIPELINE_UTIL_BOUNDING_BOX_UTILS_H_
+
+#include <limits>
+#include "dali/core/geom/box.h"
+#include "dali/core/tensor_layout.h"
+#include "dali/core/math_util.h"
+
+namespace dali {
+
+template <int ndim>
+void AxisFlip(Box<ndim, float>& box, int axis) {
+  assert(axis >= 0 && axis < ndim);
+  box.lo[axis] = 1.0f - box.hi[axis];
+  box.hi[axis] = 1.0f - box.lo[axis];
+}
+
+template <int ndim>
+void HorizontalFlip(Box<ndim, float>& box) {
+  static_assert(ndim >= 1);
```

Comment:
```suggestion
  static_assert(ndim == 1);
```
Does it make sense for empty bboxes?